x-pack/plugins/stack_connectors/server/connector_types/gen_ai/index.ts

```diff
@@ -0,0 +1,331 @@
+};
+const ConfigSchema = schema.object(configSchemaProps);
+export type ConnectorTypeConfigType = TypeOf<typeof ConfigSchema>;
+
+// secrets definition
+export type ConnectorTypeSecretsType = TypeOf<typeof SecretsSchema>;
+const secretSchemaProps = {
+  apiKey: schema.string(),
+};
+const SecretsSchema = schema.object(secretSchemaProps);
+
+// params definition
+export type ActionParamsType = TypeOf<typeof ParamsSchema>;
+const ParamsSchema = schema.object({
+  body: schema.maybe(schema.string()),
+});
+
+export const ConnectorTypeId = '.gen-ai';
+// connector type definition
+export function getConnectorType(): GenerativeAiConnectorType {
+  return {
+    id: ConnectorTypeId,
+    minimumLicenseRequired: 'gold',
+    name: i18n.translate('xpack.stackConnectors.genAi.title', {
+      defaultMessage: 'Generative AI',
+    }),
+    supportedFeatureIds: [GeneralConnectorFeatureId],
+    validate: {
+      config: {
+        schema: ConfigSchema,
+        customValidator: validateConnectorTypeConfig,
+      },
+      secrets: {
+        schema: SecretsSchema,
+      },
+      params: {
+        schema: ParamsSchema,
+      },
+    },
+    renderParameterTemplates,
+    executor,
+  };
+}
+
+function renderParameterTemplates(
+  params: ActionParamsType,
+  variables: Record<string, unknown>
+): ActionParamsType {
+  if (!params.body) return params;
+  return {
+    body: renderMustacheString(params.body, variables, 'json'),
+  };
+}
+
+function validateConnectorTypeConfig(
+  configObject: ConnectorTypeConfigType,
+  validatorServices: ValidatorServices
+) {
+  const { configurationUtilities } = validatorServices;
+  const configuredUrl = configObject.apiUrl;
+  try {
+    new URL(configuredUrl);
+  } catch (err) {
+    throw new Error(
+      i18n.translate('xpack.stackConnectors.genAi.configurationErrorApiUrl', {
+        defaultMessage: 'error configuring Generative AI action: unable to parse apiUrl: {err}',
+        values: {
+          err,
+        },
+      })
+    );
+  }
+
+  try {
+    configurationUtilities.ensureUriAllowed(configuredUrl);
+  } catch (allowListError) {
+    throw new Error(
+      i18n.translate('xpack.stackConnectors.genAi.configurationError', {
+        defaultMessage: 'error configuring Generative AI action: {message}',
+        values: {
+          message: allowListError.message,
+        },
+      })
+    );
+  }
+
+  try {
+    if (
+      configObject.apiProvider === OpenAiProviderType.OpenAi ||
+      configObject.apiProvider === OpenAiProviderType.AzureAi
+    ) {
+      return;
+    }
+    throw new Error('API Provider is not supported');
+  } catch (err) {
+    throw new Error(
+      i18n.translate('xpack.stackConnectors.genAi.configurationErrorApiProvider', {
+        defaultMessage: 'error configuring Generative AI action: invalid OpenAi Provider: {err}',
+        values: {
+          err,
+        },
+      })
+    );
+  }
+}
+
+// action executor
+export async function executor(
```

Comment:
What do you think of using the `SubActions` framework? The transition is super easy and it will make your connector "sub-action" ready for the future in case you want to support more actions. Given that you have two providers you can have one sub-action for each provider. This will help if in the future the providers diverge.

Reply:
Done!